docs/advanced-usage.rst

```diff
@@ -0,0 +1,176 @@
+Advanced Usage
+==============
+
+.. currentmodule:: urllib3
+
+.. _stream:
+
+Streaming and IO
+----------------
+
+When dealing with large responses it's often better to stream the response
+content::
+
+    >>> r = http.request(
```

Comment:
On the basic usage page, you had a snippet where you set `http = urllib3.PoolManager()`; we should probably have that again on the Advanced Usage page to reiterate that `http` isn't something that's built into urllib3.